File: cpp/41.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <iterator>
#include <utility>
#include <map>
#include <set>
#include <string>
#include <unordered_map>
#include <climits>
#include <numeric>
using namespace std;

template<typename T>
void prints(vector<T> &v){
  cout << "[";
  for(uint i=0; i<v.size()-1; ++i)
    cout << v[i] << ", ";
  if(v.size()) cout << v[v.size()-1];
  cout << "]\n";
}

class Solution {
public:
  int firstMissingPositive(int A[], int n) {
    sort(A, A+n);
    int nxt = 1;
    for(int i=0; i<n; i++)
      if(A[i] == nxt)
        ++nxt;
    return nxt;
  }
};

int main(){
  auto sol = Solution();
}

File: python/41.py
class Solution(object):
    def firstMissingPositive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.append(-1)
        n = len(nums)
        for i in range(n):
            if nums[i] != i:
                while 0 < nums[i] < n:
                    j = nums[i]
                    if nums[i] == nums[j]:
                        break
                    nums[i], nums[j] = nums[j], nums[i]
        for i in range(1, n):
            if nums[i] != i:
                return i
        return n
